iroha-lib/examples/DomainAssetCreation.cpp

```diff
@@ -0,0 +1,251 @@
+#include <iostream>
+#include <string>
+#include <cassert>
+
+#include "model/Tx.hpp"
+#include "model/Query.hpp"
+#include "model/utils/Utils.h"
+
+
+static constexpr const char* LINE = "-----------";
+
+
+iroha_lib::Query generateQueryBase(const std::string& account_name, const std::string& key_path);
+iroha::protocol::Query generateGetAccountAssetsQuery(const std::string& account_name, const std::string& key_path);
+iroha::protocol::Query generateGetAccountTransactionsQuery(const std::string& account_name, const std::string& key_path);
+iroha::protocol::Query generateGetAccountQuery(const std::string& account_name, const std::string& key_path);
+
+template<typename Tx>
+void sendTransaction(Tx& tx, const std::string& peer_ip, uint16_t torii_port);
+
+iroha::protocol::Transaction generateTransactionWhichCreatesDomainAndAsset(
+        const std::string& account_name,
+        const std::string& key_path,
+        const std::string& domain_id,
+        const std::string& user_default_role,
+        const std::string& asset_name);
+iroha::protocol::Transaction generateTransactionWhichAddsAssetQuantiti(
+        const std::string& account_name,
+        const std::string& key_path,
+        const std::string& assetIdWithDomain,
+        const std::string& assetAmount);
+
+void printAccountAssets(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port);
+void printAccount(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port);
+
+void run(const std::string& key_path);
+
+
+int main(int argc, char** argv)
```

Comment:
```suggestion
int main(const int argc, char const* const* argv)
```

Not sure if this `argv` modification will work, but let's try

Reply:
The signature that allows modifying `argc` and `argv` is standard. However, `char const* const* argv` is not required by the standard, so it might fail on exotic compilers.

Reply:
So because `int main(int argc, char** argv)` is some standard, even despite the fact that `int main(int argc, const char** argv)` would be better. I'll leave as it is and mark resolved